external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -739,4 +764,9 @@ public boolean shouldPoll() {
             return !this.pollablePartitions.isEmpty();
         }
     }
+
+    @VisibleForTesting
+    KafkaOffsetMetric getKafkaOffsetMetric() {
```

Comment:
Optional: We could avoid this getter by using the same trick with a factory as we use for injecting the KafkaConsumer. If we add a KafkaOffsetMetricFactory to the spout constructor, you can create the metric in the test and pass it into the spout instead of letting the spout construct the metric.

Reply:
Agreed,  but I feel current approach is simpler. maybe we can evolve when we add more metrics.

Reply:
Yes, makes sense. Thanks.

Reply:
If we start adding a lot of these test methods we would be better off but creating a class in the tests packaged called KafkaSpoutTest that extends KafkaSpout and use that one in the tests. All of these methods should go in this class. WE don't want this class to be very bloated.